docs/syntax-datatypes.md

```diff
@@ -122,17 +109,29 @@ User::"a1b2c3d4-e5f6-a1b2-c3d4-EXAMPLE11111"
 >principal == User::"a1b2c3d4-e5f6-a1b2-c3d4-EXAMPLE11111", // alice
 >```
 
-## IPaddr<a name="datatype-ipaddr"></a>
+## Extension<a name="datatype-extension"></a>
 
-A value that represents an IP address. It can be either IPv4 or IPv6. The value can represent an individual address or, by adding a [CIDR suffix](https://wikipedia.org/wiki/Classless_Inter-Domain_Routing#CIDR_notation) (a slash `/` and an integer) after the address.
+The remaining Cedar data types are introduced as _extension types_. Values of an extension type are introduced by calling a _constructor function_ that takes a string as its parameter. Operations on extension types, aside from equality, use a function- or method-call syntax. Equality testing uses `==` as usual.
 
-{: .note }
->You must specify values of type `IPaddr` by using the `ip()` operator. You can't specify the value as a simple literal.  
->```
->ip("192.168.1.100")
->```
-> For more information, see [ip() operator](syntax-operators.md#ip-parse-string-and-convert-to-ipaddr).
+As of now Cedar supports two extension types: [decimal](#datatype-decimal) and [ipaddr](#datatype-ipaddr).
+
+### decimal<a name="datatype-decimal"></a>
+
+A value with both a whole number part and a decimal part of no more than four digits.
+
+You specify values of extension type `decimal` by using the [`decimal()` function](syntax-operators.md#decimal-parse-string-and-convert-to-decimal), e.g.,
+```
+decimal("12345.1234")
+```
+(You can't specify a `decimal` as a simple literal.)
+
+A `decimal` value can range from -922337203685477.5808 to 922337203685477.5807.
+
+### ipaddr<a name="datatype-ipaddr"></a>
+
+A value that represents an IP address. It can be either IPv4 or IPv6. The value can represent an individual address or, by adding a [CIDR suffix](https://wikipedia.org/wiki/Classless_Inter-Domain_Routing#CIDR_notation) (a slash `/` and an integer) after the address.
```

Comment:
I think this might be an incomplete sentence? "The value can represent an individual address or [a range of addresses] by adding a CIDR suffix (a slash `/` and an integer) after the address."

Reply:
It is. This is the style that Dave adopted, and I continue to follow. If you look up at `decimal` you see it starts with the same kind of incomplete sentence. If we change it here we should change the style in the schema description and other places in the doc, too.

Reply:
I don't mind the first sentence - it's a stylistic choice. I was referring to the third sentence, which I had trouble parsing. I was proposing inserting the text in [brackets] in my comment.